fix(transformer/class-properties): create temp var for `this` in computed key

overlookmotel opened the pull request:
`this` in computed key needs a temp var, otherwise once moved into constructor it refers to the wrong `this`.

overlookmotel commented:
* **#7686** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7686?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7686?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Dunqing commented:
### Merge activity

* **Dec 5, 9:20 PM EST**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

overlookmotel committed:
fix(transformer/class-properties): create temp var for `this` in computed key (#7686)

`this` in computed key needs a temp var, otherwise once moved into constructor it refers to the wrong `this`.
